cmd/archive/unpack.go

```diff
@@ -36,62 +38,91 @@ const (
 	FailedChroot
 	FailedChdirAfterChroot
 	FailedInvokeUnpack
+	FailedApplyLayer
 )
 
-func main() {
-	ctx := context.Background()
-	op := trace.NewOperation(ctx, "Unpack") // TODO op ID?
-	op.Debugf("New unpack operation created")
+// Calls our custom unpack method inside of the chroot
+func customUnpack(op trace.Operation, unpackRoot, encodedFilterSpec string) {
+	setupChroot(op, unpackRoot)
+	op.Debugf("New custom unpack operation created")
 
-	if len(os.Args) != 4 {
-		op.Errorf("Wrong number of arguments passed to unpack binary")
-		os.Exit(WrongArgumentCount)
+	filterSpec, err := archive.DecodeFilterSpec(op, &encodedFilterSpec)
+	if err != nil {
+		op.Errorf("Couldn't deserialize filterspec %s", encodedFilterSpec)
+		os.Exit(InvalidFilterSpec)
 	}
 
-	root := os.Args[2]
+	if err = archive.UnpackNoChroot(op, os.Stdin, filterSpec, "/"); err != nil {
+		op.Error(err)
+		os.Exit(FailedInvokeUnpack)
+	}
 
-	filterSpec, err := archive.DecodeFilterSpec(op, &os.Args[3])
-	if err != nil {
-		op.Errorf("Couldn't deserialize filterspec %s", os.Args[3])
-		os.Exit(InvalidFilterSpec)
+	os.Exit(Success)
+}
+
+// Calls docker's ApplyLayer inside of the chroot
+func dockerUnpack(op trace.Operation, unpackRoot string) {
+	setupChroot(op, unpackRoot)
+	op.Debugf("New docker-based unpack operation created")
+
+	// Untar the archive
+	if _, err := docker.ApplyLayer("/", os.Stdin); err != nil {
+		op.Errorf("Error applying layer: %s", err)
+		os.Exit(FailedApplyLayer)
 	}
 
-	fi, err := os.Stat(root)
+	os.Exit(Success)
+
+}
+
+// Performs sanity checking (makes sure the unpack directory exists and has permissions, etc), changes directory to the unpack directory, and calls chroot on that directory. Will exit the executable if an error occurs.
```

Comment:
Minor: can you break this comment into multiple lines?

Reply:
yes

but also turn on line wrapping in emacs